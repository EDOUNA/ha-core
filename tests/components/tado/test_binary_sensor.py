"""The sensor tests for the tado platform."""

import logging

import aioresponses
from tadoasync import Tado

from homeassistant.const import STATE_OFF, STATE_ON
from homeassistant.core import HomeAssistant

from .util import async_init_integration

from tests.components.tado.util_new import async_init_integration_second

_LOGGER = logging.getLogger(__name__)


async def test_air_con_create_binary_sensors(hass: HomeAssistant) -> None:
    """Test creation of aircon sensors."""

    await async_init_integration(hass)

    state = hass.states.get("binary_sensor.air_conditioning_power")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.air_conditioning_connectivity")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.air_conditioning_overlay")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.air_conditioning_window")
    assert state.state == STATE_OFF


async def test_heater_create_binary_sensors(hass: HomeAssistant) -> None:
    """Test creation of heater sensors."""

    await async_init_integration(hass)

    state = hass.states.get("binary_sensor.baseboard_heater_power")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.baseboard_heater_connectivity")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.baseboard_heater_early_start")
    assert state.state == STATE_OFF

    state = hass.states.get("binary_sensor.baseboard_heater_overlay")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.baseboard_heater_window")
    assert state.state == STATE_OFF


async def test_water_heater_create_binary_sensors(hass: HomeAssistant) -> None:
    """Test creation of water heater sensors."""

    await async_init_integration(hass)

    state = hass.states.get("binary_sensor.water_heater_connectivity")
    assert state.state == STATE_ON

    state = hass.states.get("binary_sensor.water_heater_overlay")
    assert state.state == STATE_OFF

    state = hass.states.get("binary_sensor.water_heater_power")
    assert state.state == STATE_ON


async def test_home_create_binary_sensors(
    hass: HomeAssistant, python_tado: Tado, responses: aioresponses
) -> None:
    """Test creation of home binary sensors."""

    await async_init_integration_second(hass, python_tado, responses)

    for state in hass.states.async_all():
        _LOGGER.debug("Entity ID: %s, State: %s", state.entity_id, state.state)

    state = hass.states.get("binary_sensor.wr1_connection_state")
    assert state.state == STATE_ON
