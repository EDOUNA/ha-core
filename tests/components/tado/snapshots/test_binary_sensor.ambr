# serializer version: 1
# name: test_air_con_create_binary_sensors
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Air Conditioning Power',
  })
# ---
# name: test_air_con_create_binary_sensors.1
  ReadOnlyDict({
    'device_class': 'connectivity',
    'friendly_name': 'Air Conditioning Connectivity',
  })
# ---
# name: test_air_con_create_binary_sensors.2
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Air Conditioning Overlay',
    'termination': 'TADO_MODE',
  })
# ---
# name: test_air_con_create_binary_sensors.3
  ReadOnlyDict({
    'device_class': 'window',
    'friendly_name': 'Air Conditioning Window',
  })
# ---
# name: test_heater_create_binary_sensors
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Baseboard Heater Power',
  })
# ---
# name: test_heater_create_binary_sensors.1
  ReadOnlyDict({
    'device_class': 'connectivity',
    'friendly_name': 'Baseboard Heater Connectivity',
  })
# ---
# name: test_heater_create_binary_sensors.2
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Baseboard Heater Early start',
  })
# ---
# name: test_heater_create_binary_sensors.3
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Baseboard Heater Overlay',
    'termination': 'MANUAL',
  })
# ---
# name: test_heater_create_binary_sensors.4
  ReadOnlyDict({
    'device_class': 'window',
    'friendly_name': 'Baseboard Heater Window',
  })
# ---
# name: test_home_create_binary_sensors
  ReadOnlyDict({
    'device_class': 'connectivity',
    'friendly_name': 'WR1 Connection state',
  })
# ---
# name: test_water_heater_create_binary_sensors
  ReadOnlyDict({
    'device_class': 'connectivity',
    'friendly_name': 'Water Heater Connectivity',
  })
# ---
# name: test_water_heater_create_binary_sensors.1
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Water Heater Overlay',
  })
# ---
# name: test_water_heater_create_binary_sensors.2
  ReadOnlyDict({
    'device_class': 'power',
    'friendly_name': 'Water Heater Power',
  })
# ---
